src/index.js

```diff
@@ -1,5 +1,16 @@
+import { AppContainer } from 'react-hot-loader';
 import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './App';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const rootEl = document.getElementById('root');
+const render = Component =>
+  ReactDOM.render(
+    <AppContainer>
+      <Component />
+    </AppContainer>,
+    rootEl
+  );
+
+render(App);
+if (module.hot) module.hot.accept('./App', () => render(App));
```

Comment:
Yeah, that is incorrect for webpack 2, _as long as_ you disable the module transform in babel/typescript/etc. webpack has to see the raw `import` / `export` commands. The `require` is necessary if they are getting converted to `require` calls by your preprocessor.